### DISCUSSION TITLE
Add PrimarySaleHappened to CreateNftInput

### DISCUSSION THREAD

**Discussion by aaronlinsky** [metaplex-foundation/js#127] [Q&A]
For my purposes, I want to always flip the PrimarySaleHappened bit, even if the first sale hasn't happened. This is easy to do in a second transaction using `nfts().update()`, but I'd rather be able to set it from the beginning. 

Would it be possible to add PrimarySaleHappened to CreateNftInput?

**Comment by crypt0miester** [comment]
you would have to call it after create. 

there is a update primary sale happened in the token metadata program that you can use. not the update metadata.

you can do this in the same transaction btw.

**Reply by aaronlinsky** [reply]
Yup. I'm trying to use the higher level functions in the SDK, so once we can compose instructions, I'll likely just do that. 

Thanks
